op-chain-ops/interopgen/deployers/opchain.go

```diff
@@ -0,0 +1,87 @@
+package deployers
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+)
+
+type DeployOPChainInput struct {
+	OpChainProxyAdminOwner common.Address
+	SystemConfigOwner      common.Address
+	Batcher                common.Address
+	UnsafeBlockSigner      common.Address
+	Proposer               common.Address
+	Challenger             common.Address
+
+	BasefeeScalar     uint32
+	BlobBaseFeeScalar uint32
+	L2ChainId         *big.Int
+	Opsm              common.Address
+}
+
+func (input *DeployOPChainInput) InputSet() bool {
+	return true
+}
+
+type DeployOPChainOutput struct {
+	OpChainProxyAdmin                 common.Address
```

Comment:
I wonder if it makes sense to use `abigen` to generate this

Reply:
See above comment about struct API layers. And beyond the definition, the rest is already auto-generated. And the script will error with the selector of the field if it tries to set an unknown attribute, so it's easy enough to figure out if it doesn't match the ABI. Can add stricter up-front ABI checks too, like we do with the script bindings, if it's still a concern.